### ISSUE TITLE
Scene and Prefabs? or Scene and Nodes ?

### ISSUE THREAD

**Issue by Nodragem** [BabylonJS/Editor#85]
Hello,

I know that I am going very far in the road map but for now it seems that we can only create a scene and nothing else. No prefabs and no nested prefabs. 

This concept of prefab is useful in order to save reusable objects on the hard drive / asset folder, and then add them to any scenes.

However, Unity approach may not be the best. I would like to attract your attention to an alternative and maybe simpler but yet powerful way to do things: Scenes and Nodes. You can find more information in this article from Godot Engine's documentation:
http://docs.godotengine.org/en/3.0/getting_started/step_by_step/scenes_and_nodes.html

After trying Unity and Godot, I can tell that Godot approach is more fun. 
A scene can be saved on the hard drive and added to any scene.
Every scene is a tree of nodes and any branches can become a tree and so a scene. 
Hence, there are no prefabs, just scenes: a scene for your main character, a scene for the enemy, a scene for your level, a scene for the main menu.

Hope that can help opening a discussion :)

**Comment by julien-moreau** [comment]
Hey this is planned in my head but the link you shared may be really useful. I keep it in my TODO list as I would like the editor to be stable before starting this kind on feature right now :)
Anyway, I tag this issue and will let you know once I start working on it!
Thanks!

**Comment by julien-moreau** [comment]
I'm not adding this feature in the V2 milestone. But I keep the issue open in order to implement in the next step! :)

**Comment by Nodragem** [comment]
Yeah, I was also thinking that was more of a long term goal, if it becomes a goal :)

**Comment by vortice3D** [comment]
Estimado cliente/proveedor.
Hasta el próximo día 6 de Agosto me encuentro fuera de la oficina.
Durante este período tendré un acceso muy limitado al correo-e, por lo que para asuntos urgentes le ruego que se ponga en contacto con alguno de mis compañeros de trabajo, por medio del teléfono fijo de nuestras oficinas (+34 976 736 669).
Espero sepa disculpar los inconvenientes que esta situación pueda causarle.
Reciba un cordial saludo.

Dear client/provider.
At the moment I'm out of the office, and will back at August the 6th.
During this period I will have a very limited e-mail access.
For immediate assistance please contact one of my work partners, at office phone (+34 976 736 669).
I apologize for the inconveniences.
Best regards.

-------------------------------
Mario A. Martínez Latorre
+34 976 736 669
www.vortice3d.com

**Comment by Nodragem** [comment]
did you go for something similar to Godot (nodes) or Unity (prefab)?

**Comment by julien-moreau** [comment]
Good question! For performances reason and feature request reasons I used a kind of prefabs features. It is absolutely not finished (you cannot really edit prefabs for instance) and I would like to work with scenes like Godot but needs more time for me :)

**Comment by Nodragem** [comment]
Sounds good, take your time to make it great :)

**Comment by julien-moreau** [comment]
TODO: use asset containers of bjs to work as "scenes in scenes"

**Comment by julien-moreau** [comment]
Starting with prefabs from now
Opening a new issue to focus on prefabs
